cmd/cstor-pool-mgmt/volumereplica/volumereplica.go

```diff
@@ -485,3 +514,67 @@ func capacityOutputParser(output string) *apis.CStorVolumeCapacityAttr {
 	}
 	return capacity
 }
+
+// GenerateReplicaID generate new replicaID for given CVR
+func GenerateReplicaID(cvr *apis.CStorVolumeReplica) error {
+	if len(cvr.Spec.ReplicaID) != 0 {
+		return errors.Errorf("ReplicaID for cvr(%s) is already generated", cvr.Name)
+	}
+
+	csum := md5.Sum([]byte(cvr.UID))
+	cvr.Spec.ReplicaID = strings.ToUpper(hex.EncodeToString(csum[:]))
+
+	if len(cvr.Spec.ReplicaID) == 0 {
+		return errors.Errorf("ReplicaID for cvr(%s) is zero", cvr.Name)
+	}
+	return nil
+}
+
+// SetReplicaID set replicaID to volume
+func SetReplicaID(cvr *apis.CStorVolumeReplica) error {
+	var err error
+
+	vol, err := GetVolumeName(cvr)
+	if err != nil {
+		return err
+	}
+
+	ret, err := zfs.NewVolumeGetProperty().
+		WithScriptedMode(true).
+		WithParsableMode(true).
+		WithField("value").
+		WithProperty("io.openebs:zvol_replica_id").
+		WithDataset(vol).
+		Execute()
+	if err != nil {
+		return errors.Errorf("Failed to get replicaID %s", err)
+	}
+
+	sid := strings.Split(string(ret), "\n")[0]
+
+	if len(sid) == 0 {
+		lr, err := zfs.NewVolumeSetProperty().
+			WithProperty("io.openebs:zvol_replica_id", cvr.Spec.ReplicaID).
+			WithDataset(vol).
+			Execute()
+		if err != nil {
+			return errors.Errorf("Failed to set replicaID %s %s", err, string(lr))
+		}
+	}
```

Comment:
returning error would be better in 'else' case?

Reply:
error about replica_id is already set?